### PR TITLE
Added test-cases for widen in "int.jl"

### PR DESCRIPTION
**Pull request by ranaDherya**
Added few test-cases for widen in "test/int.jl"  (all the 524704 test cases of int.jl passed ).
Added tests for widen(::Type{UInt16}), widen(::Type{UInt32}) and widen(::Type{UInt64}).

### CODE REVIEW

File: test/int.jl [hunk 1]
```diff
@@ -240,6 +240,14 @@ end
     @test typeof(widen(Int64(-3))) == Int128
     @test typeof(widen(Int128(-3))) == BigInt
 
+    @test typeof(widen(UInt16(65535))) == UInt32
```
Comment (Keno) [inline test/int.jl:243]: `===` below checks type equality so these tests are redundant.

**Comment by Keno** [comment]
Is there a reason to believe the existing tests don't already adequately cover this function? (i.e. is there some missing coverage of corner cases or something?).

**Comment by ranaDherya** [comment]
@Keno 

Over here https://coveralls.io/builds/45390050/source?filename=base%2Fint.jl  its showing that "widen" for input Uint16, UInt32 and UInt64 doesnot have test coverage (these are in red colour) . so i thought that they require more tests to turn green. (I could have been wrong as its my first time in open source.) Please do tell, what I need to change?

**Comment by bramtayl** [comment]
I don't think code coverage is that reliable. There should be much more extensive testing of code coverage itself.

**Review by simeonschaub — COMMENTED** [review]
I'm sorry this PR got a little forgotten! Better test coverage of this code is definitely a great improvement. I think we can delete the explicit type checks, but otherwise we can just merge this. Thanks for the contribution!

File: test/int.jl [hunk 2]
```diff
@@ -240,6 +240,14 @@ end
...
+    @test typeof(widen(UInt32(4294967295))) == UInt64
+    @test typeof(widen(UInt64(1999999999999999990))) == UInt128
+
```
Comment (simeonschaub) [inline test/int.jl:246]:
```suggestion
```

**Comment by ranaDherya** [comment]
Created a pull request #45157 with the suggested changes. (I wasn't able to resolve merge conflicts so i deleted my fork and made the changes to a new fork and submitted it.)

> I'm sorry this PR got a little forgotten! Better test coverage of this code is definitely a great improvement. I think we can delete the explicit type checks, but otherwise we can just merge this. Thanks for the contribution!

**Pull request closed without merging**